core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -49,7 +50,10 @@ import org.apache.spark.util._
  *
  * All public methods of this class are thread-safe.
  */
-private class ShuffleStatus(numPartitions: Int) extends Logging {
+private class ShuffleStatus(
+    numPartitions: Int,
+    numReducers: Int,
```

Comment:
I'm thinking we could use `numReducers = -1` to indicate the disabling. Thus we don't need `isPushBasedShuffleEnabled`. But maybe a little bit tricky. It's up to you.

Reply:
Addressed this comment as well.